addon/services/prefetch.js

```diff
@@ -28,9 +29,24 @@ if (gte('3.6.0')) {
                 let { route, fullParams } = changeSet.for[i];
                 if (seenRoutes.has(route)) continue;
 
-                route._prefetched = new RSVP.Promise(r => {
-                  return r(route.prefetch(fullParams, transition));
-                });
+                if (isThenable(route)) {
+                  // Ensure we use the expected fullParams when the promise then
+                  // handler is executed for lazy loaded routes.
+                  RSVP.Promise.resolve(fullParams).then(capturedFullParams => {
```

Comment:
If we convert the wrapping for loop to a forEach, I think we can get rid of this funky resolve wrapping the full params

Reply:
what do you think @rwjblue , do we really need to be able to break? seems there would be a deep enough route hierarchy for it to really matter.